### PR TITLE
efficient block fetching

### PR DESCRIPTION
**Pull request by bucko13**
Update to use new utility from bpanel-utils to fetch block info with rpc call, also supports clearing the blocks table when trying to connect to a client config that doesn't work.

Update requires https://github.com/bpanel-org/bpanel-utils/pull/28

### CODE REVIEW

File: lib/components/RecentBlocks.js [hunk 1]
```diff
@@ -33,25 +33,28 @@ class RecentBlocks extends React.Component {
     // use this to ensure order of columns
     // otherwise table just uses object keys
     const colHeaders = [
```
Comment (tynes) [inline lib/components/RecentBlocks.js]: This array should only be created one time instead of in every render call, since it doesn't change

**Comment by bucko13** [comment]
Demo of changing networks between chains and networks:
demo

File: lib/components/RecentBlocks.js (hunk 1 above)
Comment (bucko13) [reply lib/components/RecentBlocks.js]: good call. updated and did the same with mainData.

File: lib/components/RecentBlocks.js [hunk 2]
```diff
@@ -8,11 +8,24 @@ import {
   ExpandedDataRow,
   Text
 } from '@bpanel/bpanel-ui';
-import { pick } from 'underscore';
+import { pick, omit } from 'underscore';
```
Comment (tynes) [inline lib/components/RecentBlocks.js]: Should we migrate to `lodash` and use it as a `peerDependency` to reduce bundle size?

Comment (bucko13) [reply lib/components/RecentBlocks.js]: yeah, good call. Missed that this was still on underscore 👍

File: lib/components/RecentBlocks.js [hunk 3]
```diff
@@ -30,32 +43,21 @@ class RecentBlocks extends React.Component {
...
-      'bits',
-      'nonce',
-      'txs'
-    ];
-
     let table;
     if (Array.isArray(recentBlocks) && recentBlocks.length) {
       // structure data for use in the expanded row component
-      recentBlocks.sort((a, b) => a - b);
+      recentBlocks.sort((a, b) => a - b).forEach(block => {
```
Comment (tynes) [inline lib/components/RecentBlocks.js]: Could you elaborate on what this code is doing exactly? Why a `forEach` instead of a `map`?

Comment (bucko13) [reply lib/components/RecentBlocks.js]: map returns a new array, this is just going through each item and updating/reformatting their values. I just didn't think it was necessary to create a new array of blocks to update the values.

I noticed that some coinbase values were coming back as arrays with the hash split into (this seems to be unique to handshake blocks actually, I should check with that team to see if there is a specific reason for it). For display purposes though the coinbase should be represented as a string.

Comment (tynes) [reply lib/components/RecentBlocks.js]: You should add comments to the code around the coinbase thing

Comment (tynes) [reply lib/components/RecentBlocks.js]: What does subtracting two objects do in the `sort` comparator?

Comment (bucko13) [reply lib/components/RecentBlocks.js]: Oh yeah, huh. Good question! `git blame` puts that as 4 months ago. That returns NaN though so not sure what I was thinking. Updating to make sure it sorts by height and adding comments.

File: lib/components/RecentBlocks.js [hunk 4]
```diff
@@ -30,32 +43,23 @@ class RecentBlocks extends React.Component {
...
     if (Array.isArray(recentBlocks) && recentBlocks.length) {
       // structure data for use in the expanded row component
-      recentBlocks.sort((a, b) => a - b);
+      recentBlocks.sort((a, b) => b.height - a.height).forEach(block => {
+        // some blocks come back with coinbase split up into an array
+        // for these situations we want to join the data for display purposes
+        block.coinbase = Array.isArray(block.coinbase)
+          ? block.coinbase.join('')
+          : block.coinbase;
+      });
```
Comment (tynes) [inline lib/components/RecentBlocks.js]: more of a personal style nit but I think that
```js
if (Array.isArray(block.coinbase))
  block.coinbase = block.coinbase.join('')
```

is way easier to read

**Pull request merged**